### ISSUE TITLE
Many warnings shows up after using Amplify code

### ISSUE THREAD

**Issue by Eumag** [aws-amplify/amplify-swift#2325]
### Describe the bug

Hello Amplify Team,

I started using Amplify.Auth functionalities and there are several warnings that shows up in its codebase. Pasted few of these warnings in Relevant log output below.

Can you check. For instance [repo](https://github.com/Eumag/app4) can be used to check and witness these warnings. I have used the steps as mentioned in [amplify-doc](https://docs.amplify.aws/lib/auth/signin/q/platform/ios/). 

I thought this might be related to some settings in Xcode or libs version, but I have verified by going through the steps mentioned in amplify-doc. 

Thanks

 

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Checkout https://github.com/Eumag/app4 and build on xcode to witness the warnings.
Or,
1. Go through Amplify link (https://docs.amplify.aws/lib/project-setup/create-application/q/platform/ios/#1-create-a-new-project) and create the project using Amplify.Auth.signIn() and related functions. 

2. Check Buildtime on Xcode (left side panel)
3. Check the yellow'ed warnings
```

### Expected behavior

Warnings should not occur, esp. not so many.

### Amplify Framework Version

1.28.0

### Amplify Categories

Auth

### Dependency manager

Cocoapods

### Swift version

5.0

### CLI version

9.2.1

### Xcode version

13.4 (13F17a)

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


/app4/Pods/AWSCore/AWSCore/Logging/AWSDDOSLogger.m:21:8: 'AWSDDOSLogger' is only available on iOS 10.0 or newer (comes from tens of files)
/app4/Pods/AWSCore/AWSCore/Bolts/AWSTask.m:19:72: This old-style function definition is not preceded by a prototype
/app4/Pods/AWSCognitoIdentityProvider/AWSCognitoIdentityProvider/AWSCognitoIdentityProviderResources.m:3092:10: Backslash and newline separated by space
/app4/Pods/AWSCore/AWSCore/Logging/AWSDDOSLogger.h:29:12: 'AWSDDOSLogger' has been marked as being introduced in iOS 10.0 here, but the deployment target is iOS 9.0.0
/app4/Pods/AWSCore/AWSCore/Logging/AWSDDOSLogger.m:21:23: Annotate 'sharedInstance' with an availability attribute to silence this warning
/app4/Pods/AWSCognitoIdentityProvider/AWSCognitoIdentityProvider/AWSCognitoIdentityUser.m:419:83: Undeclared selector 'startCustomAuthentication_v2'
...
many others
...
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

I have used these apis for first time so cannot say for sure. But I don't think such a number of warnings is a norm in aws code base.

### Device

iPhone11 - Real Device (ModelNo: MHDD3HN/A)

### iOS Version

iOS 15.6.1

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by atierian** [comment]
Thanks for opening this issue @Eumag. I believe the repo you shared is private, because I get a 404 for that link.
The majority of these warnings are related to internal Amplify code consuming other public Amplify APIs that have been deprecated; they're warnings that we're creating ourselves. That being said, you're absolutely correct and we are working towards eliminating these warnings.

**Comment by Eumag** [comment]
I am on development branch and will be making some of the experimental changes. Added you as a collaborator. If there is other access level that suits better here do let me know. If you can check to confirm anything that might have been missing from usage perspective (say in expected Target version, pods used,?), would be good to know. From my side I have used the amplify-doc mostly.

**Comment by harsh62** [comment]
@Eumag We have fixed all the warnings in Amplify 2.0.
